### PR TITLE
update cf-702C

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

AC: https://codeforces.com/contest/702/submission/292572139

### CODE REVIEW

**Comment by freakin23** [comment]
will it be right if I change the heading for the 3rd solution to `Using a Sorted List` only for Python and write the solution for it?

**Comment by freakin23** [comment]
Personally, I don't think Solution 3 is needed at all. The towers are already given in sorted order, so there is no point in using a set for that. It just looks like a heavy version of Solution 1.

File: solutions/silver/cf-702C.mdx [hunk 1]
```diff
@@ -319,10 +359,9 @@ left of the city. Define $\texttt{dist}$ to be the minimum of the distance to
 the tower on the right and the distance to the tower on the left. Then, set $r$
 to be the maximum of itself and $\texttt{dist}$.
 
-**Time Complexity:** $\mathcal{O}(N\log N)$
+**Time Complexity:** $\mathcal{O}((N + M)\log M)$
```
Comment (ryanchou-dev) [inline solutions/silver/cf-702C.mdx:362]: wouldn't this be n log m?

where's the n+m come from (input? - n, m share the same bound right)

File: solutions/silver/cf-702C.mdx [hunk 2]
```diff
@@ -137,7 +137,50 @@ public class CellularNetwork {
...
 </JavaSection>
+<PySection>
+
+```py
+"""
+returns the first index in the array that is >= value, or arr.size()
+if no such index exists
+"""
+
+
```
Comment (ryanchou-dev) [inline solutions/silver/cf-702C.mdx:148]:
```suggestion
```

File: solutions/silver/cf-702C.mdx [hunk 3]
```diff
@@ -137,7 +137,50 @@ public class CellularNetwork {
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+"""
+returns the first index in the array that is >= value, or arr.size()
```
Comment (ryanchou-dev) [inline solutions/silver/cf-702C.mdx:144]:
```suggestion
returns the first index in the array that is >= value, or len(arr)
```

File: solutions/silver/cf-702C.mdx (hunk 1 above)
Comment (freakin23) [reply solutions/silver/cf-702C.mdx:362]: how

Comment (ryanchou-dev) [reply solutions/silver/cf-702C.mdx:362]: where'd the n + m come from (input?)
bottle neck is in the for loop right 🤔

Comment (freakin23) [reply solutions/silver/cf-702C.mdx:362]: my bad I was ignoring the dominancy. thanks

Comment (freakin23) [reply solutions/silver/cf-702C.mdx]: > where'd the n + m come from (input?) bottle neck is in the for loop right 🤔

yes

File: solutions/silver/cf-702C.mdx [hunk 4]
```diff
@@ -137,7 +137,50 @@ public class CellularNetwork {
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+"""
+returns the first index in the array that is >= value, or len(arr)
+if no such index exists
+"""
```
Comment (SansPapyrus683) [inline solutions/silver/cf-702C.mdx:146]: again, doc comments go under the function

Comment (freakin23) [reply solutions/silver/cf-702C.mdx]: my bad

Comment (SansPapyrus683) [reply solutions/silver/cf-702C.mdx]: nah ur good :heart:

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**